### PR TITLE
Express notes api

### PR DESCRIPTION
**Pull request by Skim-329**
express-notes-api

### CODE REVIEW

File: express-notes-api/index.js [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+app.post('/api/notes', (req, res) => {
+  const newNote = req.body;
+  const id = data.nextId++;
+  if (newNote.content === undefined) {
+    res.status(400).json(requiredFieldError);
+  } else {
+    newNote.id = id;
+    notes[id] = newNote;
+    const fs = require('fs');
+    const r = JSON.stringify(data);
```
Comment (TimelikeClosure) [inline express-notes-api/index.js:59]: `r` isn't descriptive enough for a variable name.

**Review by TimelikeClosure — CHANGES_REQUESTED** [review]

File: express-notes-api/index.js (hunk 1 above)
Comment (Skim-329) [reply express-notes-api/index.js:59]: Change made!

**Comment by Skim-329** [comment]
Resubmit for pull request.

File: express-notes-api/index.js [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    } else {
+      delete notes[idAsNumber];
+      const returnData = JSON.stringify(data);
+
+      fs.writeFile('data.json', returnData, err => {
+        if (err) {
+          res.status(500).json(serverError);
+          throw err;
+        } else {
+          res.status(204).json();
```
Comment (scbowler) [inline express-notes-api/index.js:89]: When you don't want to send any data back you can also use `res.sendStatus(204);`

**Review by scbowler — APPROVED** [review]
Great job implementing the feedback, looks good! 👍 

1 FYI comment, no changes required.

**Pull request merged**